traversal/iterator/GraphIterator.java

```diff
@@ -268,19 +265,24 @@ private boolean isClosure(ProcedureEdge<?, ?> edge, Vertex<?, ?> fromVertex, Ver
             toIter = edge.branch(graphMgr, fromVertex, params).filter(role -> {
                 if (scoped.contains(role.asThing())) return false;
                 else {
-                    replaceScopedRole(scoped, edge, role.asThing());
-                    roles.put(edge.to().id(), role.asThing());
+                    if (scoped.orderVisited(edge.order())) scoped.replace(edge.order(), role.asThing());
+                    else scoped.push(role.asThing(), edge.order());
                     return true;
                 }
             });
+        } else if (edge.from().id().isScoped()) {
+            Identifier.Variable scope = edge.from().id().asScoped().scope();
+            Scopes.Scoped scoped = scopes.getOrInitialise(scope);
+            scoped.push(fromVertex.asThing(), edge.order());
+            toIter = edge.branch(graphMgr, fromVertex, params);
```

Comment:
we were actually forgetting to add the `from` vertex to a `scope`! This is a bug, when the start vertex is scoped to a relation that is the endpoint of an edge coming out of the start vertex.